diff --git a/inc/sendgrid.mail.inc b/inc/sendgrid.mail.inc
index 2233987..c725e0f 100644
--- a/inc/sendgrid.mail.inc
+++ b/inc/sendgrid.mail.inc
@@ -22,6 +22,9 @@ class SendGridMailSystem implements MailSystemInterface {
   public function mail(array $message) {
 
     $key_secret = variable_get('sendgrid_integration_apikey', '');
+    if (module_exists('key') && $key_value = key_get_key_value($key_secret)) {
+      $key_secret = $key_value;
+    }
     $options = array(
       'turn_off_ssl_verification' => FALSE,
       'protocol' => 'https',
diff --git a/modules/sendgrid_integration_reports/sendgrid_integration_reports.module b/modules/sendgrid_integration_reports/sendgrid_integration_reports.module
index 59643d7..5e5bb6b 100644
--- a/modules/sendgrid_integration_reports/sendgrid_integration_reports.module
+++ b/modules/sendgrid_integration_reports/sendgrid_integration_reports.module
@@ -182,6 +182,9 @@ function sendgrid_integration_reports_get_stats($cache_id, $categories = array()
   }
   // Load key from variables and throw errors if not there.
   $key_secret = variable_get('sendgrid_integration_apikey', '');
+  if (module_exists('key') && $key_value = key_get_key_value($key_secret)) {
+    $key_secret = $key_value;
+  }
   if (empty($key_secret)) {
     watchdog('sendgrid_integration_reports', t('Sendgrid Module is not setup with API key.'), array(), 'WATCHDOG_NOTICE', NULL);
     drupal_set_message(t('Sengrid Module is not setup with an API key.'), 'warning', TRUE);
@@ -290,6 +293,9 @@ function sendgrid_integration_reports_get_stats_browser() {
   }
   // Load key from variables and throw errors if not there.
   $key_secret = variable_get('sendgrid_integration_apikey', '');
+  if (module_exists('key') && $key_value = key_get_key_value($key_secret)) {
+    $key_secret = $key_value;
+  }
   if (empty($key_secret)) {
     watchdog('sendgrid_integration_reports', t('Sendgrid Module is not setup with API key.'), array(), 'WATCHDOG_NOTICE', NULL);
     drupal_set_message(t('Sengrid Module is not setup with an API key.'), 'warning', TRUE);
@@ -371,6 +377,9 @@ function sendgrid_integration_reports_get_stats_devices() {
   }
   // Load key from variables and throw errors if not there.
   $key_secret = variable_get('sendgrid_integration_apikey', '');
+  if (module_exists('key') && $key_value = key_get_key_value($key_secret)) {
+    $key_secret = $key_value;
+  }
   if (empty($key_secret)) {
     watchdog('sendgrid_integration_reports', t('Sendgrid Module is not setup with API key.'), array(), 'WATCHDOG_NOTICE', NULL);
     drupal_set_message(t('Sengrid Module is not setup with an API key.'), 'warning', TRUE);
diff --git a/plugins/key_type/sendgrid_integration_api.inc b/plugins/key_type/sendgrid_integration_api.inc
new file mode 100644
index 0000000..b165fc8
--- /dev/null
+++ b/plugins/key_type/sendgrid_integration_api.inc
@@ -0,0 +1,15 @@
+<?php
+
+/**
+ * @file
+ * Plugin definition for the Sendgrid Integration API key type.
+ */
+
+$plugin = array(
+  'label' => t('Sendgrid Integration API'),
+  'description' => t('A Sendgrid Integration API key is used to access your Sendgrid account.'),
+  'group' => 'authentication',
+  'key value' => array(
+    'plugin' => 'text_field'
+  ),
+);
diff --git a/sendgrid_integration.module b/sendgrid_integration.module
index 4ac8254..f793c34 100644
--- a/sendgrid_integration.module
+++ b/sendgrid_integration.module
@@ -67,11 +67,12 @@ function sendgrid_integration_admin() {
   );
 
   $form['authentication']['sendgrid_integration_apikey'] = array(
-    '#type' => 'password',
+    '#type' => array_key_exists('key_select', module_invoke_all('element_info')) ? 'key_select' : 'password',
     '#title' => t('API Secret Key'),
     '#description' => t('The secret key of your key pair. These are only generated once by Sendgrid.'),
     '#required' => TRUE,
     '#default_value' => variable_get('sendgrid_integration_apikey', ''),
+    '#key_filters' => array('type' => 'sendgrid_integration_api'),
   );
 
   return system_settings_form($form);
@@ -144,6 +145,9 @@ function sendgrid_integration_get_bounces($start_date, $end_date, $type = NULL,
   $server = "https://sendgrid.com/api/bounces.get.json";
   $user = variable_get('sendgrid_integration_username', '');
   $key = variable_get('sendgrid_integration_apikey', '');
+  if (module_exists('key') && $key_value = key_get_key_value($key)) {
+    $key = $key_value;
+  }
 
   if (isset($start_date) == FALSE || isset($end_date) == FALSE) {
     watchdog('SendGrid Integration', 'Error on calling get_bounches: Missing date parameter', NULL, WATCHDOG_NOTICE, $link = NULL);
