diff --git a/commerce_authnet.module b/commerce_authnet.module
index db162ef..40604d4 100644
--- a/commerce_authnet.module
+++ b/commerce_authnet.module
@@ -203,7 +203,8 @@ function commerce_authnet_aim_settings_form($settings = NULL) {
     '#required' => TRUE,
   );
   $form['tran_key'] = array(
-    '#type' => 'textfield',
+    // Use the key element, if it is available.
+    '#type' => (array_key_exists('key', module_invoke_all('element_info'))) ? 'key' : 'textfield',
     '#title' => t('Transaction Key'),
     '#description' => t('Your Transaction Key can be found on the same screen as your API Login ID. However, it will not be readily displayed. You must answer your security question and submit a form to see your Transaction Key.'),
     '#default_value' => $settings['tran_key'],
@@ -1495,6 +1496,11 @@ function commerce_authnet_aim_request($payment_method, $nvp = array()) {
     'x_solution_id' => 'A1000009',
   );
 
+// Use the Key module, if it is enabled.
+  if (module_exists('key') && $key_value = key_get_key($nvp['x_tran_key'])) {
+    $nvp['x_tran_key'] = $key_value;
+  }
+
   // Allow modules to alter parameters of the API request.
   drupal_alter('commerce_authnet_aim_request', $nvp, $payment_method);
 
