diff --git a/mandrill.admin.inc b/mandrill.admin.inc
index bb3edc5..c9c5072 100644
--- a/mandrill.admin.inc
+++ b/mandrill.admin.inc
@@ -20,10 +20,11 @@ function mandrill_admin_settings($form, &$form_state) {
   $key = variable_get('mandrill_api_key');
   $form['mandrill_api_key'] = array(
     '#title' => t('Mandrill API Key'),
-    '#type' => 'textfield',
+    '#type' => array_key_exists('key_select', module_invoke_all('element_info')) ? 'key_select' : 'textfield',
     '#description' => t('Create or grab your API key from the !link.',
       array('!link' => l(t('Mandrill settings'), 'https://mandrillapp.com/settings/index'))),
     '#default_value' => $key,
+    '#key_filters' => array('type' => 'mandrill_api'),
   );
 
   $library = libraries_detect('mandrill');
diff --git a/mandrill.module b/mandrill.module
index 0c7428f..0b40278 100644
--- a/mandrill.module
+++ b/mandrill.module
@@ -85,6 +85,9 @@ function mandrill_libraries_info() {
 function mandrill_test_access() {
   $a = user_access('administer mandrill');
   $b = variable_get('mandrill_api_key');
+  if (module_exists('key') && $key_value = key_get_key_value($b)) {
+    $b = $key_value;
+  }
   return $a & !empty($b);
 }
 
@@ -287,6 +290,9 @@ function mandrill_get_api_object($reset = FALSE) {
     }
 
     $api_key = variable_get('mandrill_api_key', '');
+    if (module_exists('key') && $key_value = key_get_key_value($api_key)) {
+      $api_key = $key_value;
+    }
     $api_timeout = variable_get('mandrill_api_timeout', 60);
     if (empty($api_key)) {
       $msg = t('Failed to load Mandrill API Key. Please check your Mandrill settings.');
diff --git a/modules/mandrill_template/mandrill_template.module b/modules/mandrill_template/mandrill_template.module
index ddb8b89..e1b9733 100644
--- a/modules/mandrill_template/mandrill_template.module
+++ b/modules/mandrill_template/mandrill_template.module
@@ -46,6 +46,9 @@ function mandrill_template_entity_info() {
 function mandrill_template_map_access() {
   $a = user_access('configure mandrill templates');
   $b = variable_get('mandrill_api_key');
+  if (module_exists('key') && $key_value = key_get_key_value($b)) {
+    $b = $key_value;
+  }
   return $a & !empty($b);
 }
 
diff --git a/plugins/key_type/mandrill_api.inc b/plugins/key_type/mandrill_api.inc
new file mode 100644
index 0000000..9bd1987
--- /dev/null
+++ b/plugins/key_type/mandrill_api.inc
@@ -0,0 +1,15 @@
+<?php
+
+/**
+ * @file
+ * Plugin definition for the Mandrill API key type.
+ */
+
+$plugin = array(
+  'label' => t('Mandrill API'),
+  'description' => t('A Mandrill API key is used to access your Mandrill account.'),
+  'group' => 'authentication',
+  'key value' => array(
+    'plugin' => 'text_field'
+  ),
+);
