diff --git a/mailgun.admin.inc b/mailgun.admin.inc
index ae42721..692d2db 100644
--- a/mailgun.admin.inc
+++ b/mailgun.admin.inc
@@ -33,6 +33,11 @@ function mailgun_admin_settings($form, &$form_state) {
     '#required' => TRUE,
   );
 
+  if (array_key_exists('key_select', module_invoke_all('element_info'))) {
+    $form['mailgun_api_key']['#type'] = 'key_select';
+    unset($form['mailgun_api_key']['#description']);
+  }
+
   // Display settings only when a valid API key is present.
   if (!empty($key)) {
     $form['mailgun_queue'] = array(
@@ -111,8 +116,11 @@ function mailgun_admin_settings($form, &$form_state) {
 function mailgun_admin_settings_validate($form, &$form_state) {
   if ($form['mailgun_api_key']['#default_value'] != $form_state['values']['mailgun_api_key']) {
     // The API key has changed. Perform validation.
-    $form_state['values']['mailgun_api_key'] = trim($form_state['values']['mailgun_api_key']);
-    $client = mailgun_get_client($form_state['values']['mailgun_api_key']);
+    $key = trim($form_state['values']['mailgun_api_key']);
+    if (module_exists('key') && $key_value = key_get_key_value($key)) {
+      $key = $key_value;
+    }
+    $client = mailgun_get_client($key);
     try {
       $result = $client->get('domains');
       drupal_set_message(t('Your API key has been successfully validated.'));
diff --git a/mailgun.module b/mailgun.module
index 40ca8f4..46cf68e 100644
--- a/mailgun.module
+++ b/mailgun.module
@@ -129,6 +129,9 @@ function mailgun_get_client($key = '') {
   }
 
   $key = (empty($key)) ? variable_get('mailgun_api_key', '') : $key;
+  if (module_exists('key') && $key_value = key_get_key_value($key)) {
+    $key = $key_value;
+  }
   if (empty($key)) {
     watchdog('mailgun', 'Mailgun client initialization failed: Missing API key.', NULL, WATCHDOG_ERROR);
     return FALSE;
@@ -166,6 +169,9 @@ function mailgun_get_client($key = '') {
  */
 function mailgun_send($mailgun_message) {
   $key = variable_get('mailgun_api_key', '');
+  if (module_exists('key') && $key_value = key_get_key_value($key)) {
+    $key = $key_value;
+  }
 
   // Extract the domain from the sender's email address. Use regular expression to check since it could be either a plain email address or in the form "Name <example@example.com>".
   $tokens = (preg_match('/^\s*(.+?)\s*<\s*([^>]+)\s*>$/', $mailgun_message['from'], $matches) === 1) ? explode('@', $matches[2]) : explode('@', $mailgun_message['from']);
